src/constructs/core/stack.ts

```diff
@@ -2,17 +2,23 @@ import type { App, StackProps } from "@aws-cdk/core";
 import { Stack, Tags } from "@aws-cdk/core";
 import { GuStackParameter, GuStageParameter } from "./parameters";
 
+export interface GuStackProps extends StackProps {
+  app?: string;
+}
+
 export class GuStack extends Stack {
   protected stage: GuStageParameter;
   protected stack: GuStackParameter;
 
-  constructor(app: App, id?: string, props?: StackProps) {
+  constructor(app: App, id?: string, props?: GuStackProps) {
     super(app, id, props);
 
     this.stage = new GuStageParameter(this);
     this.stack = new GuStackParameter(this);
 
     Tags.of(this).add("Stack", this.stack.valueAsString);
     Tags.of(this).add("Stage", this.stage.valueAsString);
+
+    props?.app && Tags.of(this).add("App", props.app);
```

Comment:
I think we'd also want to create an `app` parameter too?

Reply:
I wondered about that. Does app need to be a parameter or would you just hard code it in the app?

e.g. in `bin/cdk.ts`

```
const app = new App();
new CdkStack(app, "CdkStack", { app: "MyApp"} );
```

Reply:
Or we can add another `appParameter?: boolean` prop which, if true, would create an app parameter and use that for the tag?

Reply:
Oh that's pretty neat!

Reply:
Ah, but we should attach the value to the class, maybe with a private field and a getter? That is, we'd want a way for other constructs to refer to it, for example in a `GuLaunchConfig` I think we'd want to do something like:

```ts
class GuLaunchConfig() {
  constructor() {
    ...
    this.userData = `
     #! /usr/bin/bash
     aws s3 cp ${distBucket}/${this.stage}/${this.stack}/${this.app}.deb
    `;
  }
}
```

Maybe one for later as we understand the use case a bit more.

Reply:
Yep, I was just running into that issue. Maybe let's stick with this option only for now and I'll add a TODO which we ponder if there's a) a need to and b) a nice way to do this.

Reply:
For now, in the above example `this.app` -> `props.app` and the tag would still be free. I think that's worth it but open to thoughts otherwise...